sdks/wasm/WasmHttpMessageHandler.cs

```diff
@@ -0,0 +1,394 @@
+            fetch = (JSObject)WebAssembly.Runtime.GetGlobalObject("fetch");
+
+            // install our global hook to create a Headers object.  
+            Runtime.InvokeJS(@"
+                BINDING.mono_wasm_get_global()[""__mono_wasm_headers_hook__""] = function () { return new Headers(); }
+            ");
+
+            global = (JSObject)WebAssembly.Runtime.GetGlobalObject("");
+
+        }
+
+        protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            var tcs = new TaskCompletionSource<HttpResponseMessage>();
+            cancellationToken.Register(() => tcs.TrySetCanceled());
+
+            #pragma warning disable 4014
+            doFetch(tcs, request).ConfigureAwait(false);
+            #pragma warning restore 4014
+
+            return await tcs.Task;
+        }
+
+        private async Task doFetch(TaskCompletionSource<HttpResponseMessage> tcs, HttpRequestMessage request)
+        {
+
+            try
+            {
+                var requestObject = (JSObject)json.Invoke("parse", "{}");
+                requestObject.SetObjectProperty("method", request.Method.Method);
+                requestObject.SetObjectProperty("credentials", GetDefaultCredentialsString());
+                requestObject.SetObjectProperty("cache", GetCacheModeString());
+                requestObject.SetObjectProperty("mode", GetRequestModeString());
+
+                // We need to check for body content
+                if (request.Content != null)
+                {
+                    if (request.Content is StringContent)
+                    {
+                        requestObject.SetObjectProperty("body", await request.Content.ReadAsStringAsync());
+                    }
+                    else
+                    {
+                        requestObject.SetObjectProperty("body", await request.Content.ReadAsByteArrayAsync());
+                    }  
+                }
+
+                // Process headers
+                // Cors has it's own restrictions on headers.
+                // https://developer.mozilla.org/en-US/docs/Web/API/Headers
+                var requestHeaders = GetHeadersAsStringArray(request);
+
+                if (requestHeaders != null && requestHeaders.Length > 0)
+                {
+                    using (var jsHeaders = (JSObject)global.Invoke("__mono_wasm_headers_hook__"))
+                    {
+                        for (int i = 0; i < requestHeaders.Length; i++)
+                        {
+                            //Console.WriteLine($"append: {requestHeaders[i][0]} / {requestHeaders[i][1]}");
+                            jsHeaders.Invoke("append", requestHeaders[i][0], requestHeaders[i][1]);
+                        }
+                        requestObject.SetObjectProperty("headers", jsHeaders);
+                    }
+                }
```

Comment:
This is only trying to add to the response headers but certain headers should be added to the Content.Headers.

Reply:
Fixed